vocab-importer/src/services/processors/json_processor.py

```
@@ -0,0 +1,27 @@
+            content = await file.read()
+            data = json.loads(content.decode())
+            
+            # Validate against schema
+            validated_data = VocabularyImport(**data)
+            return validated_data.dict()
+            
+        except json.JSONDecodeError as e:
+            raise FileProcessingError(
+                f"Invalid JSON format: {str(e)}",
+                status_code=400
+            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)